pkg/crd/markers/topology.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package markers
+
+import (
+	"fmt"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+)
+
+// ToplogyMarkers list topology markers (i.e. markers that specify if a
+// list is an associative-list or a set, or if a map is atomic or not).
+var TopologyMarkers = []*definitionWithHelp{
+	must(markers.MakeDefinition("listType", markers.DescribesField, ListType(""))),
+	must(markers.MakeDefinition("listMapKey", markers.DescribesField, ListMapKey(""))),
+}
+
+func init() {
+	AllDefinitions = append(AllDefinitions, TopologyMarkers...)
+}
+
+// +controllertools:marker:generateHelp:category="CRD topology"
```

Comment:
nit: https://github.com/kubernetes-sigs/controller-tools/pull/345#discussion_r342820251

(but let's not block on that)